docs/wallet/address-tagging.md

```diff
@@ -0,0 +1,125 @@
+---
+id: address-tagging
+title: Address and UTXO tagging
+---
+
+```scala mdoc:invisible
+import org.bitcoins.core.wallet.utxo._
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.transaction._
+import org.bitcoins.core.wallet.fee._
+import org.bitcoins.core.currency._
+import org.bitcoins.wallet.models.AccountDb
+import org.bitcoins.wallet._
+
+val ExampleAddressTag = CustomAddressTag("name", "tagType")
+val exampleAddress = BitcoinAddress("1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa").get
+val account: AccountDb = null
+val destinations = Vector(TransactionOutput(Satoshis.one, exampleAddress.scriptPubKey))
+case object wallet {
+    // need to define these functions so we don't have to implement every function
+    def fundRawTransaction(
+              destinations: Vector[TransactionOutput],
+              feeRate: FeeUnit,
+              fromTagOpt: Option[AddressTag],
+              markAsReserved: Boolean) = true
+
+    def getNewAddress(tags: Vector[AddressTag]) = true
+
+    def sendToAddress(
+          address: BitcoinAddress,
+          amount: CurrencyUnit,
+          feeRate: FeeUnit,
+          fromAccount: AccountDb,
+          newTags: Vector[AddressTag]) = true
+}
+```
+
+### Using AddressTags
+
+The Bitcoin-S wallet allows you to give addresses, and their associated utxos,
+a tag. These tags allow you to separate funds between utxos so you can query utxos,
+and spend from them, based off of an AddressTag. The system also allows you to create
+your own custom address tags, that will be enforced by the library.
+
+An address tag consists of the tag name, and a tag type. We use a tag type so we can have
+tag with the same name without complications.
+
+To create an address with a tag you can use `getNewAddress` but pass in a `Vector` of `AddressTag`.
+It will add to the address tag database along with all the corresponding tags.
+
+```scala mdoc:silent
+wallet.getNewAddress(tags = Vector(ExampleAddressTag)) 
+```
+
+When sending with `sendToAddress` you can also a `Vector` of new `AddressTag`s that will be applied to the
+resulting change outputs. Any tags of a different tag type not included in `newTag`s will also be applied to
+the change outputs.
```

Comment:
doesn't have to be on this PR but I think it would be nice to have an RPC endpoint for adding a tag to an existing address

Reply:
Agreed